emperor/support_files/js/visibility-controller.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * @name VisibilityController
+ *
+ * @class Manipulates and displays the visibility of objects on screen.
+ * @property {String} [title=""] Title of the controller.
+ * @property {Node} [header=div node] jQuery element for the header
+ * which contains the uppermost elements displayed in a tab.
+ * @property {Node} [body=div node] jQuery element for the body,
+ * which contains the lowermost elements displayed in tab.
+ * This goes below the header.
+ * @property {Node} [canvas=div node] jQuery element for the canvas,
+ * which contains the header and the body.
+ * @property {Node} [container=div node] jQuery element for the parent
+ * container.
+ * This only contains the canvas.
+ * @property {Boolean} [active=false] Indicates whether the tab is front most
+ * @property {String} [identifier="EMPtab-xxxxxxx"] Unique hash identifier for
+ * the tab instance.
+ * @property {Boolean} [enabled=true] Indicates if tab can be accessed.
+ * @property {String} [description=""] Human-readable description of the tab.
+ * @property {Node} [$select=chosen node] Drop-down menu to select the metadata
+ * category to visualize by.
+ **/
+
+/*
+ * @name VisibilityController
+ *
+ * @param {Node} container, Container node to create the controller in.
+ * @params {Object} [decompViewDict] This is object is keyed by unique
+ * identifiers and the values are DecompositionView objects referring to a set
+ * of objects presented on screen. This dictionary will usually be shared by
+ * all the tabs in the application. This argument is passed by reference.
+ *
+ **/
+function VisibilityController(container, decompViewDict){
+  var helpmenu = 'Change the Visibility of the attributes on the plot, such as ' +
+                 'spheres, vectors and ellipsoids.';
+  var title = 'Visibility';
+
+  // Constant for width in slick-grid
+  var SLICK_WIDTH = 25, scope = this;
+  var name, value;
+
+  // Build the options dictionary
+  var options = {'valueUpdatedCallback':function(e, args) {
+                  var visible = args.item.value, group = args.item.plottables;
+                  VisibilityController.setPlottableAttributes(scope.decompViewDict[scope.getActiveDecompViewKey()], visible, group);
```

Comment:
Don't forget to pull out `scope.decompViewDict[scope.getActiveDecompViewKey()]` into a separate variable